index.bs

```diff
@@ -1400,6 +1400,7 @@ for a session.
 <pre class="cddl remote-cddl local-cddl">
 session.CapabilityRequest = {
   ? acceptInsecureCerts: bool,
+  ? interceptFileDialogs: bool,
```

Comment:
By having such a capability I wonder how we best separate those which belong to different protocols. Right now some of the capabilities that we have are used in both classic and bidi, but others are classic only. Now we get bidi only capabilities. All that doesn't make it easy. I wonder if we should actually re-use `unhandledPromptBehavior` for bidi as well, or if we want to make the dialog handling dependent on subscribed events.

Reply:
I think it's already separated because the BiDi spec defines these as [additional capabilities](https://w3c.github.io/webdriver-bidi/#type-session-CapabilityRequest). Or do you mean how to separate them in the CDDL definitions?

Reply:
@whimboo @jgraham What do you think? [The concept of “additional WebDriver capabilities”](https://w3c.github.io/webdriver/#dfn-additional-webdriver-capability) is defined in the Classic spec, but the WebDriver BiDi spec makes it clear that these additional WebDriver capabilities in particular are BiDi-only:

> WebDriver BiDi defines [additional WebDriver capabilities](https://w3c.github.io/webdriver/#dfn-additional-webdriver-capability). The following tables enumerates the capabilities each implementation must support for WebDriver BiDi.

Do you have a suggestion of a better way to phrase this?

Reply:
I don't think that BiDi-only capabilities are a problem in general.

Currently BiDi doesn't use `unhandledPromptBehavior` at all, but I assume that will change.

If I was designing this from scratch, I might have something like:

```
session.CapabilityRequest =  {
  …
  promptBehavior: session.PromptBehavior
}

session.PromptBehavior = {
  ?alert:  : session.PromptBehaviorOption,
  ?confirm: : session.PromptBehaviorOption,
  ?default: session.PromptBehaviorOption .default "ignore",
  ?file: : session.PromptBehaviorOption,
  ?httpAuth: : session.PromptBehaviorOption,
  ?unload: : session.PromptBehaviorOption,
}

session.PromptBehaviorOption = "accept" / "dismiss" / "ignore"
```

Then for each prompt type you could have different behaviour, depending on what you expect without needing a separate top-level capability. Classic's `unhandledPromptBehavior` would be more or less equivalent to `{default: <behavior>}`. For compatibility we could always use `unhandledPromptBehavior` as a fallback if nothing else is specified.

Reply:
It does not look like unhandledPromptBehavior in classic includes file dialogs (https://html.spec.whatwg.org/#user-prompts)? So it would not be exactly equivalent to the default behavior. Should we change to this structure as part of this PR? I am not sure how would the accept and dismiss work as I understood it's rather difficult to dismiss file dialogs also in Firefox (based on a thread in Matrix)

Reply:
Right, probably only "dismiss" and "ignore" work for all dialogs. "accept" works with dialogs which don't take a value or for which there's a reasonable default value. That doesn't include file or HTTP Auth.

Reply:
I think we should introduce something that we can extend to cover more dialogs, even if for now we only support it for file dialogs. So a definition like:
```
session.CapabilityRequest =  {
  …
  promptBehavior: session.PromptBehavior
}

session.PromptBehavior = {
  ?file: : "suppress" / "dismiss" / "none",
}
```

would be fine. `"suppress"` would emit the event but not display the dialog. `"dismiss"` would cause a cancel event to be fired, as if the Cancel button was pressed, and `"none"` would not affect the dialog (although an event would still be fired if a session was subscribed). I"d even be happy to drop `"dismiss"` for now, which I think should make it functionally equivalent to what you already have.